### PR TITLE
virtualenv refactor with fixed test in CI

### PR DESCRIPTION
**Pull request by danimtb**
- [x] Refer to the issue that supports this Pull Request: Rebases #2621
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

### CODE REVIEW

**Comment by lasote** [comment]
conflict

**Comment by niosHD** [comment]
Thank you @danimtb for pushing this PR forward!

Just a quick reminder such that it is not lost, as already [mentioned in the other review](https://github.com/conan-io/conan/pull/2621#issuecomment-383535224), the used `is_posix` check only works for the msys subsystem that I used during development. The generation of this flag needs to be re-evaluated based on the information I collected in #2638 to properly implement the behavior requested in #2568.

File: conans/client/generators/virtualenv.py [hunk 1]
```diff
@@ -16,104 +16,120 @@ def __init__(self, conanfile):
...
+        :return: placeholder for the variable name formatted for a certain execution environment.
+        (e.g., cmd, ps1, sh).
+        """
+        if flavor == "cmd":
+            return "%%%s%%" % name
+        if flavor == "ps1":
+            return "$env:%s" % name
+        return "$%s" % name  # flavor == sh
+
+    def format_values(self, flavor, variables=None, env=None):
```
Comment (lasote) [inline conans/client/generators/virtualenv.py]: make "variables" positional argument, pass it always. Remove not used "env"

File: conans/test/generators/generators_test.py [hunk 2]
```diff
@@ -35,11 +37,16 @@ def test_base(self):
         client = TestClient()
         client.save(files)
         client.run("install . --build")
+
+        virtualenvFiles = ["activate.sh", "deactivate.sh"]
```
Comment (memsharded) [inline conans/test/generators/generators_test.py]: pep8 variable name

**Pull request merged**